app/feature/feature-terminate-insurance/src/main/kotlin/com/hedvig/android/feature/terminateinsurance/data/GetContractsToTerminateUseCase.kt

```diff
@@ -0,0 +1,63 @@
+package com.hedvig.android.feature.terminateinsurance.data
+
+import arrow.core.Either
+import arrow.core.raise.either
+import com.apollographql.apollo3.ApolloClient
+import com.hedvig.android.apollo.safeExecute
+import com.hedvig.android.apollo.toEither
+import com.hedvig.android.core.common.ErrorMessage
+import com.hedvig.android.core.uidata.UiMoney
+import com.hedvig.android.data.contract.ContractGroup
+import com.hedvig.android.data.contract.toContractGroup
+import com.hedvig.android.logger.logcat
+import kotlinx.datetime.LocalDate
+import octopus.ContractsToTerminateQuery
+
+internal interface GetContractsToTerminateUseCase {
+  suspend fun invoke(): Either<ErrorMessage, List<InsuranceForCancellation>>
+}
+
+internal class GetContractsToTerminateUseCaseImpl(
+  private val apolloClient: ApolloClient,
+) : GetContractsToTerminateUseCase {
+  override suspend fun invoke(): Either<ErrorMessage, List<InsuranceForCancellation>> {
+    logcat { "mariia: GetContractsToTerminateUseCaseImpl launched" }
+    return either {
+      val member = apolloClient
+        .query(ContractsToTerminateQuery())
+        .safeExecute()
+        .toEither(::ErrorMessage)
+        .bind()
+        .currentMember
+      member.toInsurancesForCancellation()
+    }
+  }
+}
+
+data class InsuranceForCancellation(
```

Comment:
Same logic here, this could also be `TerminatableInsurance` perhaps?
I'd also just avoid adding the name `Cancel` instead of `terminate` because then both of those concepts exist in the same place making it confusing if one means something else compared to the other, while both in our context mean termination.

In the same logic as above, we call them `Insurance` here, while we call them `Contract` above. Perhaps sticking with `Contract` in both would lower the mental overhead of reading those, wdyt?